gui/storage/forms.py

```diff
@@ -130,6 +130,12 @@ def _clean_zfssize_fields(form, attrs, prefix):
     for attr in attrs:
         formfield = '%s%s' % (prefix, attr)
         match = r.match(cdata[formfield].replace(' ', ''))
+        try:
+            int(cdata[formfield])
+            continue
+        except ValueError:
+            pass
+
```

Comment:
No, it will throw a ValueError if there is 10M.

When merely a int is provided we can continue to the next item.